server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -639,4 +641,395 @@ private TrackingTool toTrackingTool(CRTrackingTool crTrackingTool) {
+        if (timerConfig == null)
+            return null;
+        String spec =  timerConfig.getTimerSpec();
+        if (StringUtils.isBlank(spec))
+            throw new RuntimeException("timer schedule is not specified");
+        return new CRTimer(spec, timerConfig.shouldTriggerOnlyOnChanges());
+    }
+
+    public CREnvironmentVariable environmentVariableConfigToCREnvironmentVariable(EnvironmentVariableConfig environmentVariableConfig) {
+        if (environmentVariableConfig.isSecure()) {
+            return new CREnvironmentVariable(environmentVariableConfig.getName(), null, environmentVariableConfig.getEncryptedValue());
+        } else {
+            String value = environmentVariableConfig.getValue();
+            if(StringUtils.isBlank(value))
+                value = "";
+            return new CREnvironmentVariable(environmentVariableConfig.getName(), value);
+        }
+    }
+
+    private CRPluggableScmMaterial pluggableScmMaterialConfigToCRPluggableScmMaterial(PluggableSCMMaterialConfig pluggableScmMaterialConfig) {
+        SCMs scms = getSCMs();
+        String id = pluggableScmMaterialConfig.getScmId();
+        SCM scmConfig = scms.find(id);
+        if (scmConfig == null)
+            throw new ConfigConvertionException(
+                    String.format("Failed to find referenced scm '%s'", id));
+
+        return new CRPluggableScmMaterial(pluggableScmMaterialConfig.getName().toString(),
+                id, pluggableScmMaterialConfig.getFolder(),
+                pluggableScmMaterialConfig.getFilterAsString());
+    }
+
+    public CRPackageMaterial packageMaterialToCRPackageMaterial(PackageMaterialConfig packageMaterialConfig) {
+        return new CRPackageMaterial(packageMaterialConfig.getName().toString(), packageMaterialConfig.getPackageId());
+    }
+
+    public CRDependencyMaterial dependencyMaterialConfigToCRDependencyMaterial(DependencyMaterialConfig dependencyMaterialConfig) {
+        CRDependencyMaterial crDependencyMaterial = new CRDependencyMaterial(
+                dependencyMaterialConfig.getPipelineName().toString(),
+                dependencyMaterialConfig.getStageName().toString());
+        if (dependencyMaterialConfig.getName() != null)
+            crDependencyMaterial.setName(dependencyMaterialConfig.getName().toString());
+        return crDependencyMaterial;
+    }
+
+    private CRScmMaterial scmMaterialToCRScmMaterial(ScmMaterialConfig scmMaterialConfig) {
+        String materialName = null;
+        if (scmMaterialConfig.getName() != null) {
+            materialName = scmMaterialConfig.getName().toString();
+        }
+        if (scmMaterialConfig instanceof GitMaterialConfig) {
+            GitMaterialConfig git = (GitMaterialConfig) scmMaterialConfig;
+            return new CRGitMaterial(materialName, git.getFolder(), git.isAutoUpdate(), git.isShallowClone(), git.getUrl(), git.getBranch(), git.isInvertFilter(), git.getFilterAsString());
```

Comment:
Does `git.getFilterAsString()` really work? I'm not completely sure, but it's unlikely to be correct. The filters will have to be separate. Same with other materials. If it works in one place, I guess it works elsewhere.